cpp/src/gandiva/precompiled/CMakeLists.txt

```diff
@@ -125,10 +124,7 @@ if(ARROW_BUILD_TESTS)
                  ../decimal_xlarge.cc)
   target_include_directories(gandiva-precompiled-test PRIVATE ${CMAKE_SOURCE_DIR}/src)
   target_link_libraries(gandiva-precompiled-test PRIVATE ${ARROW_TEST_LINK_LIBS} RE2::re2)
-  target_compile_definitions(gandiva-precompiled-test
-                             PRIVATE
-                             GANDIVA_UNIT_TEST=1
-                             ARROW_STATIC
+  target_compile_definitions(gandiva-precompiled-test PRIVATE GANDIVA_UNIT_TEST=1
```

Comment:
This is to pass the build in windows. Because of ARROW_STATIC flag, in the included arrow library headers the ARROW_EXPORT doesn't resolve to __declspec(dllimport) and so linker can't find the definitions during linking.  
ARROW_STATIC flag seems to be added for compiling the vendor/datetime/tz.cpp; since we are linking the arrow library so it should already find the definitions there, so removing both.